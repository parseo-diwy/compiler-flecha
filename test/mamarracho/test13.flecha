
def test numero fn valor =
  unsafePrintChar 'T';
  unsafePrintChar 'E';
  unsafePrintChar 'S';
  unsafePrintChar 'T';
  unsafePrintChar ' ';
  unsafePrintInt numero;
  unsafePrintChar '\n';
  unsafePrintChar '\t';
  fn valor;
  unsafePrintChar '\n'

def pb b      = unsafePrintChar (if b then 'T' else 'F')
def and b1 b2 = if b1 then b2 else False
def or  b1 b2 = if b1 then True else b2
def not b     = if b then False else True 

def main =
  test 0 pb (and True True);
  test 1 pb (and True False);
  test 2 pb (and False True);
  test 3 pb (and False False);
  test 4 pb (or True True);
  test 5 pb (or True False);
  test 6 pb (or False True);
  test 7 pb (or False False);
  test 8 pb (not True);
  test 9 pb (not False)

