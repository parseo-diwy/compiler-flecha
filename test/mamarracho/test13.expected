TEST 0
	T
TEST 1
	F
TEST 2
	F
TEST 3
	F
TEST 4
	T
TEST 5
	T
TEST 6
	T
TEST 7
	F
TEST 8
	F
TEST 9
	T
